### PR TITLE
MSM base changes for CUDA and ASM

### PR DESCRIPTION
**Pull request by Protryon**
This PR does some minor refactors for CUDA and ASM, which will be rebased on this PR.

### CODE REVIEW

File: algorithms/src/msm/variable_base/mod.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+use snarkvm_curves::traits::AffineCurve;
+use snarkvm_fields::{PrimeField, Zero};
+
+mod standard;
+
+pub struct VariableBaseMSM;
+
+impl VariableBaseMSM {
+    #[allow(unused)]
```
Comment (howardwu) [inline algorithms/src/msm/variable_base/mod.rs:25]: Can you use a `deprecated` flag here so it is not used?

File: algorithms/src/msm/variable_base/mod.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    fn msm_basic<G: AffineCurve>(bases: &[G], scalars: &[<G::ScalarField as PrimeField>::BigInteger]) -> G::Projective {
```
Comment (howardwu) [inline algorithms/src/msm/variable_base/mod.rs:26]: Can you rename to `msm_naive`?

File: algorithms/src/msm/variable_base/mod.rs (hunk 1 above)
Comment (Protryon) [reply algorithms/src/msm/variable_base/mod.rs:25]: It's a private(unexposed) method, and serves as a reference implementation for tests that is more likely to be correct due to simplicity. It is used in tests. I think `deprecated` flag is not appropriate, if that's fine with you.

Comment (Protryon) [reply algorithms/src/msm/variable_base/mod.rs:25]: Was also renamed to `naive` rather than `basic`, which I think is more appropriate.

Comment (howardwu) [reply algorithms/src/msm/variable_base/mod.rs:25]: Just realized it's private. Can you move it below so it's clearly private?

**Review by howardwu — APPROVED** [review]

**Pull request merged**